tools/ALARAJOYWrapper/preprocess_fendl3.py

```diff
@@ -45,20 +45,62 @@ def remove_gas_daughters(all_rxns):
                     }
                 }    
             }
-    
+        gas_tuples (list of tuples): Pairs total gas production MT values with
+            their respective gas symbols of the form [(gas, MT), ...].
+ 
     Returns:
-        filtered_rxns (collections.defaultdict): Modified version of all_rxns
-            with double-counted gas-producing reactions left out.
+        all_rxns (collections.defaultdict): Modified version of all_rxns with
+            double-counted gas-producing reactions left out.
     """
 
-    gas_tuples = list(tp.GAS_DF[['kza', 'total_mt']].itertuples(index=False))
     for parent in all_rxns:
         for gKZA, gMT in gas_tuples:
             if gKZA in all_rxns[parent] and gMT in all_rxns[parent][gKZA]:
                 all_rxns[parent][gKZA] = {gMT: all_rxns[parent][gKZA][gMT]}
 
     return all_rxns
 
+def subtract_gas_from_totals(all_rxns, gas_tuples):
+    """
+    For any reaction that produces a gas daughter, subtract the individual
+        cross-sections from the list of total gas production cross sections
+        corresponding to MT = 203-207. Optional method to be called within
+        gas_handling().
+
+    Arguments:
+        all_rxns (collections.defaultdict): Hierarchical dictionary keyed by
+            parent nuclides to store all reaction data, with structured as:
+            {parent:
+                {daughter:
+                    {MT:
+                        {
+                            'emitted': (str of emitted particles)
+                            'non_zero_groups': (int of non-zero groupwise XS)
+                            'xsections': (array of groupwise XS)
+                        }
+                    }
+                }    
+            }
+        gas_tuples (list of tuples): Pairs total gas production MT values with
+            their respective gas symbols of the form [(gas, MT), ...].
+    
+    Returns:
+        all_rxns (collections.defaultdict): Modified version of all_rxns with
+            each reaction that produces a gas daughter calculating its cross-
+            sections through subtraction from gas production totals.
+
+    """
+
+    for parent in all_rxns:
+        for gKZA, gMT in gas_tuples:
+            mt_data = all_rxns[parent][gKZA]
```

Comment:
We don't know if `gKZA` is in `all_rxns[parent]` yet - we test that on the next line, so might be easiest to use the full variable like we do in `remove_gas_daughters`